Doc/library/test.rst

```diff
@@ -1419,3 +1419,21 @@ script execution tests.
    containing the *source*.  If *compiled* is ``True``, both source files will
    be compiled and added to the zip package.  Return a tuple of the full zip
    path and the archive name for the zip file.
+
+.. function:: catch_unraisable_exception()
+
+   Context manager catching unraisable exception using
+   :func:`sys.unraisablehook`.
+
+   Usage::
+
+       with support.catch_unraisable_exception() as cm:
+           # code creating an "unraisable exception"
```

Comment:
Might be worth suggesting gc.collect()

Reply:
gc.collect() is not needed. Why do you want to force a GC collection?

Reply:
It's not needed but it's often a good trigger of these sort of exceptions

Reply:
Well, tests using the context manager *expect* an unraisable exception and fail if it's not the case.

Reply:
Oh. By the way. I also had the idea of installing a custom sys.unraisablehook in regrtest and repeat all errors at exit. Maybe even make the test fail if at least one unraisable exception is logged.

Reply:
Can you URL me to that commit?

Reply:
It's not a commit but an idea :-) I didn't implement it.

Reply:
Ok, :$ ping me when implemented!